subsys/net/lib/lwm2m/lwm2m_senml_cbor.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/*
+ * Generated using cddl_gen version 0.3.99
+ * https://github.com/NordicSemiconductor/cddl-gen
+ * Generated with a default_max_qty of 3
+ */
+
+#ifndef LWM2M_SENML_CBOR_H__
+#define LWM2M_SENML_CBOR_H__
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+#include <cbor_common.h>
+#include <cbor_decode.h>
+#include <cbor_encode.h>
+
+struct record_bn {
+	cbor_string_type_t hndl;
+};
+
+struct record_bt {
+	int32_t val;
+};
+
+struct record_n {
+	cbor_string_type_t hndl;
+};
+
+struct record_t {
+	int32_t val;
+};
+
+struct record_union_ {
+	union {
+		struct {
+			int32_t v; /** Value */
+		};
```

Comment:
Why these single values are inside anonymous structures?

Reply:
That's what the code generator creates. I try not to look there as much as possible.